html/semantics/scripting-1/the-script-element/moving-between-documents.html

```diff
@@ -0,0 +1,142 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Moving script elements between documents</title>
+<link rel="author" href="mailto:d@domenic.me" title="Domenic Denicola">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/#execute-the-script-block">
+
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<!-- Background:
+
+- https://www.w3.org/Bugs/Public/show_bug.cgi?id=11323
+- https://github.com/whatwg/html/issues/2137
+- https://github.com/whatwg/html/issues/2469
+- https://github.com/whatwg/html/pull/2673
+
+-->
+
+<body>
+<script>
+"use strict";
+window.didExecute = false;
+
+for (const type of ["text/javascript", "module"]) {
+  async_test(t => {
+    t.add_cleanup(() => {
+      window.didExecute = false;
+    });
+
+    const iframe = document.createElement("iframe");
+    iframe.onload = t.step_func_done(() => {
+      iframe.contentWindow.didExecute = false;
+      iframe.contentDocument.write("<streaming-element>");
+      document.body.appendChild(iframe.contentDocument.querySelector("streaming-element"));
+      iframe.contentDocument.write(`<script id="s1" type="${type}">window.didExecute = true;<` + "/script>");
+      iframe.contentDocument.write("</streaming-element>");
+
+      const s = document.querySelector("#s1");
+      s.onload = t.unreached_func("onload");
+      s.onerror = t.unreached_func("onerror");
+
+      iframe.contentDocument.close();
+
+      assert_false(window.didExecute, "The script must not have executed in this window");
+      assert_equals(iframe.contentWindow.didExecute, undefined,
```

Comment:
Even if the script tested ("s1") is not executed, the test fails on Chromium, because didExecute is `false` (not `undefined`).
`iframe.contentWindow.didExecute` is set to `false` at Line 32 and is reset to undefined when document.write() is called on Firefox (but not on Chromium) at Line 33.

Minimized case:
```
<body>
<script>
    const iframe = document.createElement("iframe");
    iframe.onload = () => {
      iframe.contentWindow.didExecute = false;
      iframe.contentDocument.write("<div>");
      console.log(iframe.contentWindow.didExecute);
    };
    document.body.appendChild(iframe);
</script>
</body>
```
console log: `undefined` on Firefox and `false` on Chromium.